packages/fuels-core/src/code_gen/functions_gen.rs

```diff
@@ -96,15 +104,20 @@ pub fn generate_script_main_function(
 
     Ok(quote! {
         #doc
-        pub async fn #name(&self #(,#arg_declarations)*) -> Result<Vec<Receipt>, SDKError> {
+        pub async fn #name(&self #(,#arg_declarations)*) -> Result<#output_type, SDKError> {
             let provider = self.wallet.get_provider().expect("Provider not set up").clone();
-            let tokens = [#(#arg_names.into_token()),*];
-            let script_data = ABIEncoder::encode(&tokens)?.resolve(0);
-            Ok(run_script_binary(self.binary_filepath.as_str(),
-                                    Some(TxParameters::default()),
-                                    Some(provider),
-                                    Some(script_data)).await?
+            let arg_name_tokens = [#(#arg_names.into_token()),*];
+            let script_data = ABIEncoder::encode(&arg_name_tokens)?.resolve(0);
+            let mut receipts = run_script_binary(self.binary_filepath.as_str(),
+                                                    Some(TxParameters::default()),
+                                                    Some(provider),
+                                                    Some(script_data)
             )
+            .await?;
+            let output_param = #output_type::param_type();
```

Comment:
If the `output_type` turns out to be a type with generic parameters then this is not the correct syntax for calling a method associated with such a type. Rather it would look something like this:
```rust
SomeType::<GenericA,GenericB...>::param_type();
```
This is handled already in the `single_param_type_call` fn.